tests/components/zha/test_cover.py

```diff
@@ -129,3 +158,151 @@ async def get_chan_attr(*args, **kwargs):
     # test rejoin
     await async_test_rejoin(hass, zigpy_cover_device, [cluster], (1,))
     assert hass.states.get(entity_id).state == STATE_OPEN
+
+
+async def test_shade(hass, zha_device_joined_restored, zigpy_shade_device):
+    """Test zha cover platform for shade device type."""
+
+    # load up cover domain
+    zha_device = await zha_device_joined_restored(zigpy_shade_device)
+
+    cluster_on_off = zigpy_shade_device.endpoints.get(1).on_off
+    cluster_level = zigpy_shade_device.endpoints.get(1).level
+    entity_id = await find_entity_id(DOMAIN, zha_device, hass)
+    assert entity_id is not None
+
+    # test that the cover was created and that it is unavailable
+    assert hass.states.get(entity_id).state == STATE_UNAVAILABLE
+
+    # allow traffic to flow through the gateway and device
+    await async_enable_traffic(hass, [zha_device])
+    await hass.async_block_till_done()
+
+    # test that the state has changed from unavailable to off
+    await send_attributes_report(hass, cluster_on_off, {8: 0, 0: False, 1: 1})
+    assert hass.states.get(entity_id).state == STATE_CLOSED
+
+    # test to see if it opens
+    await send_attributes_report(hass, cluster_on_off, {8: 0, 0: True, 1: 1})
+    assert hass.states.get(entity_id).state == STATE_OPEN
+
+    # close from UI command fails
+    with patch("zigpy.zcl.Cluster.request", side_effect=asyncio.TimeoutError):
+        await hass.services.async_call(
+            DOMAIN, SERVICE_CLOSE_COVER, {"entity_id": entity_id}, blocking=True
+        )
+        assert cluster_on_off.request.call_count == 1
+        assert cluster_on_off.request.call_args[0][0] is False
+        assert cluster_on_off.request.call_args[0][1] == 0x0000
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5575421)